core/server/api/v2/subscribers.js

```diff
@@ -0,0 +1,203 @@
+const _ = require('lodash');
+const fs = require('fs-extra');
+const models = require('../../models');
+const fsLib = require('../../lib/fs');
+const common = require('../../lib/common');
+
+const subscribers = {
+    docName: 'subscribers',
+    browse: {
+        options: [
+            'page',
+            'limit',
+            'fields',
+            'filter',
+            'order',
+            'debug'
+        ],
+        permissions: true,
+        query(frame) {
+            return models.Subscriber.findPage(frame.options);
+        }
+    },
+
+    read: {
+        headers: {},
+        data: [
+            'id',
+            'email'
+        ],
+        permissions: true,
+        query(frame) {
+            return models.Subscriber.findOne(frame.options, _.omit(frame.options, ['data']));
+        }
+    },
+
+    add: {
+        statusCode: 201,
+        headers: {},
+        permissions: true,
+        query(frame) {
+            return models.Subscriber.getByEmail(frame.data.subscribers[0].email)
+                .then((subscriber) => {
+                    if (subscriber && frame.options.context.external) {
+                        // we don't expose this information
+                        return Promise.resolve(subscriber);
+                    } else if (subscriber) {
+                        return Promise.reject(new common.errors.ValidationError({message: common.i18n.t('errors.api.subscribers.subscriberAlreadyExists')}));
+                    }
+
+                    return models.Subscriber
+                        .add(frame.data.subscribers[0], _.omit(frame.options, ['data']))    // Q: do we still need this omit here?
```

Comment:
Do we still need the omit here since options and data are separated on the frame level?